src/controllers/accomodationController.js

```diff
@@ -76,6 +79,8 @@ const doUpdate = async (req, res) => {
       accommodation.name = req.body.name;
     }
     if (req.body.location) {
+      const location = changeToArray(req.body.location);
+      req.body.location = location;
       accommodation.locationId = req.body.location;
     }
```

Comment:
you can directly assign accommodation.locationId to location without changing req.body.location